jar-connector/src/main/java/com/sixsq/slipstream/connector/ConnectorBase.java

```diff
@@ -294,18 +294,31 @@ protected String getPrivateSshKeyFileName() throws ConfigurationException, Valid
         return privateSshKeyFile;
     }
 
+    public static String getServerPublicSshKeyFilename() throws ConfigurationException, ValidationException {
+		return Configuration.getInstance().getProperty(ServiceConfiguration.CLOUD_CONNECTOR_ORCHESTRATOR_PUBLICSSHKEY);
+    }
+
     private String getUserPublicSshKey(User user) throws ValidationException, IOException {
 		return user.getParameter(
 				ExecutionControlUserParametersFactory.CATEGORY + "." + UserParametersFactoryBase.SSHKEY_PARAMETER_NAME)
 				.getValue();
     }
 
+    private String getUserOrServerPublicSshKey(User user) throws ValidationException, IOException {
```

Comment:
Ok, I agree.
It should be possible to not add any public key but this will require a change on all connectors.
I created an issue for that.